nix/mobile/android/default.nix

```diff
@@ -1,9 +1,25 @@
-{ pkgs, deps, callPackage, mkShell
-, jsbundle, status-go, androidPkgs, androidShell }:
+{ pkgs, deps, lib, callPackage, mkShell
+, jsbundle, status-go-func, androidPkgs, androidShell }:
 
-rec {
+let
+  # Android architectures to build for
+  # Used to detect end-to-end builds
+  androidAbiInclude = lib.getEnvWithDefault "ANDROID_ABI_INCLUDE" "armeabi-v7a;arm64-v8a;x86";
+
+  # Map ANDROID_ABI_INCLUDE to status-go targets
+  androidAbiIncludeSplit = lib.splitString ";" androidAbiInclude;
+  archMap = {
+      "armeabi-v7a" = "android/arm";
+      "arm64-v8a" = "android/arm64";
+      "x86" = "android/386";
+      "x86_64" = "android/386";
+    };
+  targets = builtins.map (arch: builtins.getAttr arch archMap) androidAbiIncludeSplit;
```

Comment:
This is the right approach, but maybe in the wrong place? Lets think about what we have:

* `nix/mobile/android/release.nix` - Android build derivation (which should be called `build.nix` really).
* `nix/mobile/android/default.nix` - Wrapper around Android build derivation.
* `nix/status-go/mobile/build.nix` - The `status-go` mobile build derivation.
* `nix/status-go/mobile/default.nix` - Wrapper around `status-go` mobile build derivation.

Now, what is the purpose of those:

* `nix/mobile/android/release.nix` - Hide complexity of Android build.
* `nix/mobile/android/default.nix` - Hide complexity of creating Android dev shells from Android build.
* `nix/status-go/mobile/build.nix` - Hide complexity of `status-go` mobile build.
* `nix/status-go/mobile/default.nix` - Hide complexity of different `status-go` builds for different platforms.

Based on that purpose I'd say the right place for this architecture mapping is `nix/status-go/mobile/default.nix`.
Something like:
```nix
  android = { abis ? ["armeabi-v7a" "arm64-v8a"] }:
    callPackage ./build.nix {
      platform = "android";
      platformVersion = "23";
      outputFileName = "status-go-${source.shortRev}.aar";
      # Hide different arch naming in gomobile from Android builds.
      targets = let
        abiMap = {
          "armeabi-v7a" = "android/arm";
          "arm64-v8a"   = "android/arm64";
          "x86"         = "android/386";
          "x86_64"      = "android/386";
        };
      in map (arch: abiMap."${arch}") abis;
      inherit meta source goBuildLdFlags;
    };
```
What do you think?